### DISCUSSION TITLE
Payment method preferences returns different object types depending on setup

### DISCUSSION THREAD

**Discussion by seand7565** [solidusio/solidus#4584] [New Features or Ideas]
When setting up a payment_method, if you don't use static model preferences, you get a Hash object when you call `payment_method.preferences`. If you _do_ use static model preferences, you instead get a `Spree::Preferences::StaticModelPreferences::Definition` object when you call `payment_method.preferences`

While using the default preference source, this command will succeed:
`payment_method.preferences[:available_to_users]`
While using static model preferences, the same command will fail.

**Solidus Version**:
2.10 - though likely an issue in more versions

**To Reproduce**
1. Create a Check payment method
2. Set it to use a static model preference (example provided below)
3. Open a console, run `Spree::PaymentMethod.last.preferences[:test_mode]` - it will fail
4. Set the payment method to use the "custom" preference source
5. In the console, run `Spree::PaymentMethod.last.preferences[:test_mode]` - it will succeed

Example static model preference:
```ruby
# config/initializers/example_credentials.rb

Spree::Config.configure do |config|
  config.static_model_preferences.add(
    Spree::PaymentMethod::Check,
    'example_credential_source', {
      test_mode: false
    }
  )
end
```

**Current behavior**
`payment_method.preferences` could return either a Hash or a model object depending on the setup

**Expected behavior**
`payment_method.preferences` should always return just one object type - IMO preferably a Hash object.

**Screenshots**
![image](https://user-images.githubusercontent.com/5720486/88084963-c6b52480-cb4a-11ea-842c-8d4fc53d848c.png)
![image](https://user-images.githubusercontent.com/5720486/88084972-c87ee800-cb4a-11ea-9409-cb42b3f1b420.png)

**Comment by peterberkenbosch** [comment]
Using `preferred_` prefix for the preferences works on both setups. I do agree the hash option is more Ruby idiomatic :)
